Not connected to frontend yet.

TuckerL-OSU opened the pull request:
Uses Deprecated resoruces.
Currently it masks /employees behind the admin login, and then a normal user sees that page as 404 to protect it. A normal user can still see pages likes /awards.

Not connected to frontend yet.
I was only testing it on 8080 through the backend, I hadn't added the front end yet. Spring auto generated the the /login page, and that is where you should get the cookie. The cookie should in theory be passed to the browser when accessing it through our website on 4200 though.

tmrietz committed:
Add an extensible logging service
  - Write logInsert, logUpdate, logDelete, entityDiff methods
  - Add example service for AwardType to demonstrate how to use
  - Minor table and model changes to support

Remove my client dir since it was renamed

TuckerL-OSU committed:
Uses Deprecated resoruces.
Currently it masks /employees behind the admin login, and then a normal user sees that page as 404 to protect it. A normal user can still see pages likes /awards.

Not connected to frontend yet.

TuckerL-OSU committed:
Uses Deprecated resoruces.
Currently it masks /employees behind the admin login, and then a normal user sees that page as 404 to protect it. A normal user can still see pages likes /awards.

Not connected to frontend yet.

tmrietz committed:
Tweaks to log table schema to support logging

TuckerL-OSU committed:
Added Sessions to front and back.

***erp-server***
/api/admin/** is hidden behind admin account login
/api/** should only be accessable by user accounts
Added success and failure login handlers
Added unauth handler

IMPORTANT!! - Played with the mapping of endpoints in UserAccountController. Will probably fix this later on.

UserManager stuff is to manage sessions.

***erp-client***
When testing I added some lazy logic for getting awards. Will remove at a later time.

Created a credentials model for holding username and password.
Created a user model for holding user id, email, username, and maybe isAdmin.

Added logic to authentication.service.ts for authentication, login, isAuthenticated, and logout.
Currently need to do /api/logout to logout. Adding a button soon.

If an endpoint seems broken, try changing the proxy.conf file to be /** instead of /api/**

TuckerL-OSU committed:
Added Sessions to front and back.

***erp-server***
/api/admin/** is hidden behind admin account login
/api/** should only be accessable by user accounts
Added success and failure login handlers
Added unauth handler

IMPORTANT!! - Played with the mapping of endpoints in UserAccountController. Will probably fix this later on.

UserManager stuff is to manage sessions.

***erp-client***
When testing I added some lazy logic for getting awards. Will remove at a later time.

Created a credentials model for holding username and password.
Created a user model for holding user id, email, username, and maybe isAdmin.

Added logic to authentication.service.ts for authentication, login, isAuthenticated, and logout.
Currently need to do /api/logout to logout. Adding a button soon.

If an endpoint seems broken, try changing the proxy.conf file to be /** instead of /api/**

taylor-jones committed:
Updated employee region relationship models and integrated log service with Awards

taylor-jones committed:
Made sure every model has equals(), hashcode(), and toString(). Fixed Log model getters

taylor-jones committed:
Updated UserAccount controller and service to integrate w/ LogService

tmrietz committed:
Rietz/log work (#10)

* Add an extensible logging service
  - Write logInsert, logUpdate, logDelete, entityDiff methods
  - Add example service for AwardType to demonstrate how to use
  - Minor table and model changes to support

Remove my client dir since it was renamed

* Tweaks to log table schema to support logging

tmrietz committed:
Add a schema dump script (#14)

Should help with keeping track of schema changes. Considering doing the same for data as well.

Edit:
  - Assumes the existence of a few new environment variables

taylor-jones committed:
Fixed merge conflicts

taylor-jones committed:
Added Employee service and controller and updated cascades for relationships

taylor-jones committed:
Added Region service and controller and removed region naturalId

taylor-jones committed:
Added endpoint to get all awards for employee

taylor-jones committed:
Updated ability to find awards for user, awardType, or employee

taylor-jones committed:
Added GET /{id}/regions endpoint

taylor-jones committed:
Added endpoints to find all employees for a region and all awards for a region

taylor-jones committed:
Merge pull request #16 from taylor-jones/jones/users-and-awards

Jones/users and awards

taylor-jones committed:
Updated DML & DDL to include user_account.enabled

taylor-jones committed:
Updated user_account model to include enabled field

TuckerL-OSU committed:
Added Sessions to front and back, works with our new slightly modified db.

***erp-server***
/api/admin/** is hidden behind admin account login
/api/** should only be accessable by user accounts
Added success and failure login handlers
Added unauth handler

Fixed endpoints I broke with final commit to this branch.

UserManager stuff is to manage sessions.

***erp-client***
When testing I added some lazy logic for getting awards. Will remove at a later time.

Created a credentials model for holding username and password.
Created a user model for holding user id, email, username, and maybe isAdmin.

Added logic to authentication.service.ts for authentication, login, isAuthenticated, and logout.
Currently need to do /api/logout to logout. Adding a button soon.

TuckerL-OSU committed:
Added Sessions to front and back, works with our new slightly modified db.

***erp-server***
/api/admin/** is hidden behind admin account login
/api/** should only be accessable by user accounts
Added success and failure login handlers
Added unauth handler

Fixed endpoints I broke with final commit to this branch.

UserManager stuff is to manage sessions.

***erp-client***
When testing I added some lazy logic for getting awards. Will remove at a later time.

Created a credentials model for holding username and password.
Created a user model for holding user id, email, username, and maybe isAdmin.

Added logic to authentication.service.ts for authentication, login, isAuthenticated, and logout.
Currently need to do /api/logout to logout. Adding a button soon.

tmrietz commented:
Will have to rebase off of dev/master and fix merge conflicts before this can go in.

taylor-jones committed:
Merge pull request #17 from taylor-jones/jones/updated-user-account

Jones/updated user account

TuckerL-OSU committed:
Added Sessions to front and back.

***erp-server***
/api/admin/** is hidden behind admin account login
/api/users/whoAmI is callback for checking login status
/api/users/me WIP maybe for getting users profile?
/api/** should only be accessable by user accounts
Added success and failure login handlers
Added unauth handler

IMPORTANT!! - Played with the mapping of endpoints in UserAccountController. Will probably fix this later on.

UserManager stuff is to manage sessions.

***erp-client***
When testing I added some lazy logic for getting awards. Will remove at a later time.

Created a credentials model for holding username and password.
Created a user model for holding user id, email, username, and maybe isAdmin.

Added logic to authentication.service.ts for authentication, login, isAuthenticated, and logout.
Currently need to do /api/logout to logout. Adding a button soon.

TuckerL-OSU committed:
Added Sessions to front and back.

***erp-server***
/api/admin/** is hidden behind admin account login
/api/users/whoAmI is callback for checking login status
/api/users/me WIP maybe for getting users profile?
/api/** should only be accessable by user accounts
Added success and failure login handlers
Added unauth handler

IMPORTANT!! - Played with the mapping of endpoints in UserAccountController. Will probably fix this later on.

UserManager stuff is to manage sessions.

***erp-client***
When testing I added some lazy logic for getting awards. Will remove at a later time.

Created a credentials model for holding username and password.
Created a user model for holding user id, email, username, and maybe isAdmin.

Added logic to authentication.service.ts for authentication, login, isAuthenticated, and logout.
Currently need to do /api/logout to logout. Adding a button soon.

TuckerL-OSU committed:
Merge remote-tracking branch 'origin/lavellt-sessions-cookies' into lavellt-sessions-cookies
